src/ai/ai.cpp

```diff
@@ -70,6 +76,11 @@ void update_ai_general_status(sys::state& state) {
 		bool threatened = defensive_str < safety_factor * greatest_neighbor;
 		state.world.nation_set_ai_is_threatened(n, threatened);
 
+		// provinces owned by rebels -> immediate threat
+		for(const auto po : n.get_province_ownership_as_nation())
+			if(bool(po.get_province().get_province_rebel_control().get_rebel_faction()))
+				state.world.nation_set_ai_is_threatened(n, true);
```

Comment:
This will make the ai accept alliances when occupied by a rebel somewhere, but those allies will not help it out

Reply:
If you want the ai to do different things when rebels are activated, that is fine, but you cannot use the is_threatened flag for it